compiler/ballerina-lang/src/main/java/io/ballerina/projects/util/FileUtils.java

```diff
@@ -285,23 +288,49 @@ public static void deleteDeprecatedMetaFile(Path metaFilePath) {
         }
     }
 
+    public static void replaceTemplateName(Path path, String templateName, String packageName) {
+        Optional<Path> fileName = Optional.ofNullable(path.getFileName());
+        if (fileName.isPresent() && fileName.get().toString().endsWith(BLANG_SOURCE_EXT)) {
```

Comment:
Thinking out loud; shouldn't we throw an error if the given bal file is not existing?